AudioPlayer/AudioPlayer/Content/AudioPlayer/AudioPlayerControlsView.swift

```diff
@@ -247,11 +248,23 @@ extension AudioPlayerControls {
         func play(_ track: AudioTrack) {
             if track != currentTrack {
                 currentTrack?.status = .idle
-                audioPlayerService.play(url: track.url)
-                currentTrack = track
+                if track.url.scheme == "custom" {
+                    let source = createStreamSource()
+                    let audioFormat = AVAudioFormat(
+                        commonFormat: .pcmFormatFloat32, sampleRate: 44100, channels: 2, interleaved: false
+                    )!
+                    audioPlayerService.play(source: source, entryId: track.url.absoluteString, format: audioFormat)
+                    currentTrack = track
+                } else {
+                    audioPlayerService.play(url: track.url)
+                }
             }
         }
 
+        func createStreamSource() -> CoreAudioStreamSource {
+            return CustomStreamAudioSource(underlyingQueue: audioPlayerService.player.sourceQueue)
```

Comment:
I don't like this bit, passing the inner sourceQueue from the AudioPlayer itself, but I know why it's required.
Can't think of a quick way of getting this from the AudioPlayer.